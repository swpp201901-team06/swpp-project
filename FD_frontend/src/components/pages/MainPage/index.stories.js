import React from 'react'
import { storiesOf } from '@kadira/storybook'
import { MainPage } from 'components'

storiesOf('MainPage', module)
  .add('default', () => (
    <MainPage />
  ))
