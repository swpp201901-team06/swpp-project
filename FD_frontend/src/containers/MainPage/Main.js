import { connect } from 'react-redux'
import { Main } from '../../components/pages/MainPage/Molecules/Main'
import { goToSignin, goToGuest } from '../../store/Mainpage/actions'
import { baseHistory } from '../../index'

const mapStateToProps = (state) =>{
    return {
        statefunction : state
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onClickSignin: () => {
            baseHistory.push('/signin')
            dispatch(goToSignin())
        },
        onClickGuest: () => {
            baseHistory.push('/guest')
            dispatch(goToGuest())
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Main)
