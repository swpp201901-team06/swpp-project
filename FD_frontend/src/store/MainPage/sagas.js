import { take, fork } from 'redux-saga/effects'
import { baseHistory } from '../../index'
import * as actions from './actions'

export function* watchGoToSignin() {
  while (true) {
    yield take(actions.GOTO_SIGN_IN)
    baseHistory.push('/signin')
  }
}

export function* watchGoToGuest() {
  while (true) {
    yield take(actions.GOTO_GUEST)
    baseHistory.push('/guest')
  }
}

export default function* () {
  yield fork(watchGoToSignin)
  yield fork(watchGoToGuest)
}