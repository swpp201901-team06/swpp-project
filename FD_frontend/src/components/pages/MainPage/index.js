import React from 'react'
import Main from '../../../containers/MainPage/Main'

import { PageTemplate, Header, Hero, Footer, FeatureList } from 'components'

const MainPage = () => {
  return (
    <div>
        <Main/>
    </div>
  )
}

export default MainPage
