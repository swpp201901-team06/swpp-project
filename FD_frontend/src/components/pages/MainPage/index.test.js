import React from 'react'
import { shallow } from 'enzyme'
import MainPage from './index'


describe('MainPage component test', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<MainPage />)
  })

  it('matches snapshot', () => {
    const MainPageComponent = wrapper
    expect(MainPageComponent).toMatchSnapshot()
  })

  it('renders MainPage', () => {
    expect(wrapper.length).toEqual(1)
  })

  test('renders Main molecule', () => {
    expect(wrapper.exists('Main')).toBe(true)
  })
})

