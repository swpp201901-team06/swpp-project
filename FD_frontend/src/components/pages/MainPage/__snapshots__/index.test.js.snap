// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component test matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.div>
        <img
                src={
                        Object {
                                "default": "file",
                              }
                }
                style={
                        Object {
                                "height": 200,
                                "margin": 100,
                                "width": 500,
                              }
                }
        />
        <Connect(Main)
                style={
                        Object {
                                "height": "30%",
                                "width": "30%",
                              }
                }
        />
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            src={
                        Object {
                                    "default": "file",
                                  }
            }
            style={
                        Object {
                                    "height": 200,
                                    "margin": 100,
                                    "width": 500,
                                  }
            }
/>,
          <Connect(Main)
            style={
                        Object {
                                    "height": "30%",
                                    "width": "30%",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": Object {
              "default": "file",
            },
            "style": Object {
              "height": 200,
              "margin": 100,
              "width": 500,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "height": "30%",
              "width": "30%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.div>
          <img
                    src={
                              Object {
                                        "default": "file",
                                      }
                    }
                    style={
                              Object {
                                        "height": 200,
                                        "margin": 100,
                                        "width": 500,
                                      }
                    }
          />
          <Connect(Main)
                    style={
                              Object {
                                        "height": "30%",
                                        "width": "30%",
                                      }
                    }
          />
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <img
              src={
                            Object {
                                          "default": "file",
                                        }
              }
              style={
                            Object {
                                          "height": 200,
                                          "margin": 100,
                                          "width": 500,
                                        }
              }
/>,
            <Connect(Main)
              style={
                            Object {
                                          "height": "30%",
                                          "width": "30%",
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": Object {
                "default": "file",
              },
              "style": Object {
                "height": 200,
                "margin": 100,
                "width": 500,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "height": "30%",
                "width": "30%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
